apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/index.jsx

```diff
@@ -70,13 +76,29 @@ export default function ProviderRate({
   );
 
   const setDexRate = useCallback(
-    provider => {
+    (provider) => {
       setDexSelected(provider);
       updateSelection(provider);
     },
     [updateSelection],
   );
 
+  useEffect(() => {
+    if (filter.includes(FILTER.decentralised)) {
+      setDexRate(DEX_PROVIDERS[0]);
+    } else {
+      let selectedRate;
+      if (filter.includes(FILTER.float)) {
+        selectedRate = rates.find((rate) => rate.tradeMethod === FILTER.float);
+      } else if (filter.includes(FILTER.fixed)) {
+        selectedRate = rates.find((rate) => rate.tradeMethod === FILTER.fixed);
+      }
+      setRate(selectedRate || (rates && rates[0]) || {});
+    }
+    setEmptyState(!providerRef.current?.children.length);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
any chance rates is changing over time? I imagine you only care about the "initial rates" then? as you need to actively ignore the dep?